### ISSUE TITLE
Standalone Kubelet node validation

### ISSUE THREAD

**Issue by dchen1107** [kubernetes/kubernetes#14783]
cAdvisor has built Jenkins e2e project to validate standalone cAdvisor and docker containers on different OS distro: Debian, Ubuntu, CoreOS, etc. We should extend this to including Kubelet, cAdvisor, and Docker or other container runtime. 

cc/ @kubernetes/goog-node

**Comment by dchen1107** [comment]
cc/ @quinton-hoole

**Comment by dchen1107** [comment]
@ixdy and @vishh is migrating today's cAdvisor-only Jenkins project to Kubernetes existing one, so that we can have single dashboard.

**Comment by dchen1107** [comment]
@pwittrock Could you please provide latest updates for this?

**Comment by pwittrock** [comment]
tl;dr: cadvisor+heapster have been moved, and a kubelet e2e ci has been setup.  Things are pretty good, but work to continue to improve the job flakiness and visibility is required.

Milestones:
- cadvisor + heapster CI moved to k8s jenkins CI project
  - In progress PR to convert to yaml #21489 - blocked on decision as to if we put the docker credentials id in github.com (not actual credentials, just the id)
- cadvisor + heapster PR moved to k8s jenkins PR project
- cadvisor e2e reduced flakiness
  - use ssh directly instead of gcloud compute to `gcloud compute ssh` flakiness
  - whitelist retryable known flakes
- kubelet e2e test implemented and running on CI
  - running against several different os distros and docker versions
  - In progress PR to provision instances from images instead of using static host list #21204

In progress:
- Refactor node e2e tests so they are trivial to run locally
- Refactor node e2e tests so they are less prone to `gcloud compute` flakes
- Setup node e2e test PR builder - blocked on technical issues preventing canarying a PR builder (it is all or nothing)

Future work:
- Move node e2e to critical builds once the PR and CI jobs have proven stable
- Better accounting of which osdistro/dockerversion environments are covered by testing
- Provide mechanism for defining osdistro owners and allowing them to test & publish results (instead of internally testing in private Google gce project)
